lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -32,384 +57,165 @@ def get(section: str, option: str, fallback=None) -> str:
         return "TO-INDEX, TO-COPY-SOS"
 
 
+@dataclass()
+class StatResult:
+    st_mode: int
+    st_size: int
+
+
+class MockPath:
+
+    fails: list[str] = []
+
+    def __init__(self, path: Union[str, Path, "MockPath"]):
+        self.path = str(path)
+
+    def resolve(self, strict: bool = False):
+        assert strict
+        for t in datasets:
+            if self.path == t.tarball:
+                if t.dataset.resource_id in __class__.fails:
+                    break  # Drop out to raise an exception
+                return MockPath(self.path)
+        raise FileNotFoundError(f"No such file or directory: '{self.path}'")
+
+    def stat(self) -> StatResult:
+        for t in datasets:
+            if self.path == t.tarball:
+                return StatResult(st_mode=0o554, st_size=t.size)
+        raise FileNotFoundError(f"No such file or directory: '{self.path}'")
+
+    @classmethod
+    def _fail_on(cls, fails: list[str]):
+        cls.fails.extend(fails)
+
+    @classmethod
+    def _reset(cls):
+        cls.fails.clear()
+
+
+class MockSync:
+
+    record: dict[str, JSONOBJECT] = {}
+
+    def __init__(self, logger: Logger, component: str):
+        self.component = component
+        self.logger = logger
+
+    def next(self, operation: Operation) -> list[Dataset]:
+        return [x.dataset for x in datasets]
+
+    def update(self, dataset: Dataset, did: Operation, enabled: list[Operation]):
+        __class__.record[dataset.resource_id] = {"did": did, "enabled": enabled}
+
+    @classmethod
+    def _reset(cls):
+        cls.record = {}
+
+
+class MockCacheManager:
+
+    fails: list[str] = []
+    unpacked: list[str] = []
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.config = config
+        self.logger = logger
+
+    def unpack(self, id: str):
+        if id in __class__.fails:
+            for d in datasets:
+                if d.dataset.resource_id == id:
+                    raise TarballUnpackError(Path(d.tarball), "test error")
+            else:
+                raise Exception("I'm lost; where are you?")
```

Comment:
Perhaps we should raise something more specific than `Exception`...like `AssertionError` or `NotImplementedError` or `ValueError`...or `DatasetNotFound`?